backend/src/ee/services/dynamic-secret/dynamic-secret-service.ts

```diff
@@ -164,28 +168,51 @@ export const dynamicSecretServiceFactory = ({
         throw new BadRequestError({ message: "Provided dynamic secret already exist under the folder" });
     }
 
+    let dynamicSecretInputConfig = "";
     const selectedProvider = dynamicSecretProviders[dynamicSecretCfg.type as DynamicSecretProviders];
-    const decryptedStoredInput = JSON.parse(
-      infisicalSymmetricDecrypt({
+    const { encryptor: secretManagerEncryptor, decryptor: secretManagerDecryptor } =
+      await kmsService.createCipherPairWithDataKey({
+        type: KmsDataKey.SecretManager,
+        projectId
+      });
+    if (
+      dynamicSecretCfg.keyEncoding &&
+      dynamicSecretCfg.inputCiphertext &&
+      dynamicSecretCfg.inputTag &&
+      dynamicSecretCfg.inputIV
+    ) {
+      dynamicSecretInputConfig = infisicalSymmetricDecrypt({
```

Comment:
Hmmm why i didn't go for it because i had to introduce all those logics here, and also my concern is size of the dataset as we have to reencrypt all those ones. But let me see if i can do this

Reply:
Has changed to table migation